setup.py

```diff
@@ -13,9 +13,16 @@
 # governing permissions and limitations under the License.
 
 from setuptools import setup, find_packages
+import os
+
+v = open('haas/version.py')
+exec(v.read())
```

Comment:
Response to "1. We should be careful about how we're locating the file;"
I looked around on the Internet for best practices and realized that the most practiced method is to have setup.py in the root folder of the project. Hence, the relative path should work just fine. 
https://packaging.python.org/en/latest/distributing.html#setup-py

Response to "2. is the exec necessary? Is there a reason we couldn't just import it?"
Importing the package in the setup.py may invoke a race condition and circular dependency both of which can be a problem for new users trying to install the package. Circular dependency arises because the package will try to import all other modules, many of which may not have been installed. 

I came across some posts on stackoverflow, like this one:
http://stackoverflow.com/questions/2058802/how-can-i-get-the-version-defined-in-setup-py-setuptools-in-my-package
Thus to avoid the potential issues of race condition and also make the code compatible across python 2.X and 3.X I implemented the solution as submitted.